### ISSUE TITLE
Framer Motion example skips exit animations

### ISSUE THREAD

**Issue by carloitaben** [TanStack/router#2635]
### Which project does this relate to?

Router

### Describe the bug

Hello! Loving the library so far! 😄

I was trying some of the examples and it seems that Framer Motion exit animations are not playing on the `with-framer-motion` example. Next.js has this problem as well. One of the workarounds found by the community is [this one](https://github.com/vercel/next.js/issues/49279#issuecomment-1675393849). I was wondering if the issue in Router could be solved by doing something similar 🤔

Maybe transitioning between pages with `AnimatePresence` is not worth the hassle? In Next.js there's also [this library](https://github.com/ismamz/next-transition-router) which intercepts navigations by using a custom `Link` component and `useRouter` hook. It plays an animation before commiting the navigation. This method doesn't work when using the browser's back and forwards buttons tho, but maybe that's more of a Next.js limitation

### Your Example Website or App

https://stackblitz.com/github/tanstack/router/tree/main/examples/react/with-framer-motion?embed=1&theme=dark&preset=node&file=src/main.tsx

### Steps to Reproduce the Bug or Issue

1. From the index route, click on the about link
2. Observe how the index component instantly unmounts, without playing the exit animation

### Expected behavior

It should unmount after playing the exit animation

### Screenshots or Videos

https://github.com/user-attachments/assets/d545fca8-e418-4f04-923a-1d6eccbcb077

### Platform

N/A

**Comment by klapacz** [comment]
Hey, I wrote a [blog post](https://klapacz.dev/blog/0003-tanstack-router-animated-modals/) on using View Transitions for exit animations :smile:.

**Comment by akashlama98** [comment]
And no.. Animations done with View Transitions are not interruptible like motion.dev animation. So, View Transitions isn't a fix. I hope tanstack team fixes this. This is the only thing stopping me from migrating from react-router, because in react-router I wrap AnimatePresence with `useOutlet()` hook instead of `<Outlet />` and it just works. I hope this gets fixed 🤞

**Comment by klapacz** [comment]
Yes, this was a solution for my use cases. I shared it here because it might also help others. I understand that it doesn’t help you, and I also hope there will be a solution that works for you.

**Comment by SheathedBlade** [comment]
I resonate with this so much! I used tanstack early on, but after wanted to use animations. Was disappointed that AnimatePresence was super hacky, and View Transitions did not give me satisfactory results. Here's hoping they fix this
